File: AkitaModel/SmartScan/Events/FileEvent.cs
namespace Skyware.Lis.AkitaModel.SmartScan.Events;


/// <summary>
/// Represents a file uploaded to BLOB storage.
/// This is an event payload object (SignalR).
/// </summary>
public class FileEvent : FileBase, ITypedEvent
{

    /// <summary>
    /// Id of the attachment type
    /// </summary>
    public int DocumentTypeId { get; set; }

    /// <inheritdoc/>
    public string MethodName => "file";

}

File: AkitaMotelTS/src/SmartScan/Events/FileEvent.ts
/**
 * Skyware.Lis.AkitaModel.SmartScan.Events
 */

import { FileBase } from '../FileBase';
import { ITypedEvent } from './ITypedEvent';

/**
 * Represents a file uploaded to BLOB storage.
 * This is an event payload object (SignalR).
 */
export class FileEvent extends FileBase implements ITypedEvent {
  /**
   * Id of the attachment type
   */
  documentTypeId!: number;

  /** @inheritdoc */
  get methodName(): string {
    return 'file';
  }
}
